tests/features/pep695_test.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import annotations
+
+import sys
+
+import pytest
+
+from add_trailing_comma._main import _fix_src
+
+
+@pytest.mark.parametrize(
+    's',
+    (
+        pytest.param(
+            'class A[K]:\n'
+            '    def __init__(self, value: K)\n'
+            '        pass\n',
+            id='single line classdef',
+        ),
+        pytest.param(
+            'def notNone[K](v: K) -> K:\n'
+            '    if v is None:\n'
+            '        raise ValueError("shouldn\'t be None")\n'
+            '    return v\n',
+            id='single line functiondef',
+        ),
+        pytest.param(
+            'type ListOrSet[T] = list[T] | set[T]',
+            id='single line generic type alias',
+        ),
+        pytest.param(
+            'type ListOrSet = list[str] | set[int]',
+            id='no type-param type alias',
+        ),
+    ),
+)
+def test_noop(s):
+    assert _fix_src(s) == s
+
+
+@pytest.mark.xfail(sys.version_info < (3, 12), reason='py312+')
+@pytest.mark.parametrize(
+    ('s', 'e'),
+    (
+        pytest.param(
+            'class ClassA[\n'
+            '    T: str\n'
+            ']:\n'
+            '    def method1(self) -> T:\n'
+            '        ...',
+
+            'class ClassA[\n'
+            '    T: str,\n'
+            ']:\n'
+            '    def method1(self) -> T:\n'
+            '        ...',
+            id='multiline classdef',
+        ),
+        pytest.param(
+            'def f[\n'
+            '    T\n'
+            '](x: T) -> T:\n'
+            '    return x',
+
+            'def f[\n'
+            '    T,\n'
+            '](x: T) -> T:\n'
+            '    return x',
+            id='multiline functiondef',
+        ),
+        pytest.param(
+            'type ListOrSet[\n'
+            '    T,\n'
+            '    K\n'
+            '] = list[T] | set[K]',
+            'type ListOrSet[\n'
+            '    T,\n'
+            '    K,\n'
+            '] = list[T] | set[K]',
+            id='multiline generic type alias',
+        ),
+        pytest.param(
+            'def f[\n'
+            '    T: (\n'
+            '        "ForwardReference",\n'
+            '        bytes\n'
+            '    )\n'
+            '](x: T) -> T:\n'
+            '    return x',
+
+            'def f[\n'
+            '    T: (\n'
+            '        "ForwardReference",\n'
+            '        bytes,\n'
+            '    ),\n'
```

Comment:
this isn't a valid type annotation as written

Reply:
The PEP has the following example:
```python
class ClassB[T: ("ForwardReference", bytes)]: ...  # OK
```
in the [following sectiton](https://peps.python.org/pep-0695/#constrained-type-specification) and the grammar seems to be changing the same way for [both functions and classes](https://peps.python.org/pep-0695/#grammar-changes)

Reply:
ah neat -- I was wondering how they handled the constrained types -- cool